### PR TITLE
Add metric functions tests

### PR DESCRIPTION
**Pull request by andrewelamb**

### CODE REVIEW

**Comment by andrewelamb** [comment]
@thomasyu888 @allaway

This is not ready to merge, but please comment

File: R/scoring_functions.R [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+#' @export
+c_statistic <- function(gold, pred){
+  Hmisc::rcorr.cens(pred, survival::Surv(gold), outx = TRUE)["C Index"]
+}
+
+#' Calculate AUC for a set of binary outcomes and a numeric vector of
+#'prediction values.
+#' @param gold A binary vector of outcome values.
+#' @param pred A numeric vector of prediction values.
+auc <- function(gold, pred){
```
Comment (thomasyu888) [inline R/scoring_functions.R:105]: Duplicated, we have `score_auc`

**Review by thomasyu888 — CHANGES_REQUESTED** [review]

**Comment by thomasyu888** [comment]
Can also merge `develop` back in because this branch has all the validation functions

**Comment by thomasyu888** [comment]
I merged my CI in, so the tests will auto run now for you when pushing to the branch.   If you're curious to learn about github actions, here is the action that is set to run on PR into `develop` https://github.com/Sage-Bionetworks/challengescoring/blob/develop/.github/workflows/r_cmd_check.yml

**Comment by andrewelamb** [comment]
It looks like check is failing. That's annoying bucause it could be the tests, but it could be any of the other checks that are specific to getting it accepted to CRAN.

**Comment by andrewelamb** [comment]
package e1071 is required
Backtrace:
 1. testthat::expect_equal(...)
 4. challengescoring::score_mcc(...)
##[error]Error: R CMD check found ERRORs
Execution halted
 7. caret:::confusionMatrix.default(data = pred, reference = gold)
 8. getFromNamespace("confusionMatrix.table", "caret")(...)
 9. caret:::requireNamespaceQuietStop("e1071")

══ testthat results  ═══════════════════════════════════════════════════════════
[ OK: 40 | SKIPPED: 0 | WARNINGS: 0 | FAILED: 1 ]
1. Error: score_mcc (@test-scoring_functions.R#35)

**Comment by andrewelamb** [comment]
e1071 is a dependency of caret, I wonder why it's not being installed in the test env.

**Comment by andrewelamb** [comment]
I added e1701 to the description file and that seems to have fixed the error. I also removed the duplicate auc function.

**Comment by thomasyu888** [comment]
The test command itself runs really quickly, and I've set it to ignore most things except when it errors out on the tests.  

It may be annoying but it seems like it was useful

**Pull request closed without merging**